ligo/src/types.mligo

```diff
@@ -0,0 +1,230 @@
+// Corresponds to Types.hs module
+
+// -- FA2 types -- //
+
+#if !TYPES_H
+#define TYPES_H
```

Comment:
Since the only way for one module in LIGO to depend on another module is C-like inlining (literally inlining to form one big file), we can import the same module several times and have to deal with duplication, and the recommended way to do this is again C-like preprocessor magic. This is described e.g. on [this fancy page](https://tezosacademy.io/reason/chapter-preprocessor), and I think this applies to any syntax.

Usually, it is possible to inline the same module multiple times without problems, but in our case we have sum types in `types.mligo`, and so importing module twice ends with "conflicting constructor name".

Reply:
Oh I see, so this is a "if it was imported already don't import it again" kinda thing. Got it

P.s. that's indeed a fancy page :sweat_smile: